bool action
float32 x
float32 y